src/main/java/baza/trainee/service/impl/EventServiceImpl.java

```diff
@@ -33,10 +40,19 @@ public Event getById(String id) {
     }
 
     @Override
-    public Event save(EventPublication newEvent) {
-        Event event = eventMapper.toEvent(newEvent);
+    public Event save(EventPublication newEvent, String sessionId) {
+        Event eventToSave = eventMapper.toEvent(newEvent);
+        Event savedEvent = eventRepository.save(eventToSave);
+
+        var fileNames = newEvent.content().stream()
+                .filter(cb -> cb.getBlockType().equals(BlockType.PICTURE_BLOCK)
+                || cb.getBlockType().equals(BlockType.PICTURE_TEXT_BLOCK))
+                .map(cb-> cb.getPictureLink())
+                .collect(Collectors.toList());
+
+        imageService.persist(fileNames, sessionId);
 
-        return eventRepository.save(event);
+        return eventRepository.save(savedEvent);
```

Comment:
Why do we save the Event twice?

Reply:
Now save only eventToSave
return eventRepository.save(eventToSave);